### PR TITLE
K8s YouTube vanity URL

### PR DESCRIPTION
**Pull request by chris-short**
Hello! Every week I typically have to figure out one URL or another to get people to the YouTube channel. Creating this redirect would make my life easier. The @kubernetes/youtube-admins team might agree too.

Signed-off-by: Chris Short <chris@chrisshort.net>

### CODE REVIEW

File: k8s.io/configmap-nginx.yaml [hunk 1]
```diff
@@ -289,4 +289,13 @@ data:
           rewrite ^/.*$  https://prow.k8s.io/tide redirect;
         }
       }
+
+      server {
+        server_name youtube.k8s.io youtube.kubernetes.io;
```
Comment (dims) [inline k8s.io/configmap-nginx.yaml]: do you want to do `go.k8s.io/youtube` instead? (lines 229-248)

Comment (chris-short) [reply k8s.io/configmap-nginx.yaml]: youtube.k8s.io is actually shorter.

Comment (chris-short) [reply k8s.io/configmap-nginx.yaml]: This is a precedent setter though I guess, no? Was there a preferred format that I missed?

**Comment by mrbobbytables** [comment]
Holding for @castrojo to weigh in (he's out of the office but will be back next week). As far as I know, we previously had a single url to use for community meetings but switched to a generated link for each one. I don't recall the specific reason for it however.

/hold

**Comment by chris-short** [comment]
This is for the overarching channel. Not an individual meeting. If memory serves, @castrojo didn't want to have to manage a vanity URL with the Community meeting itself because the URL does change with every meeting. Since this is the high-level channel, that's not a concern here.

**Comment by mrbobbytables** [comment]
> This is for the overarching channel. Not an individual meeting. If memory serves, @castrojo didn't want to have to manage a vanity URL with the Community meeting itself because the URL does change with every meeting. Since this is the high-level channel, that's not a concern here.

Ahh, sorry -- I thought that's what you were getting at with tracking down the url every week.

**Comment by mrbobbytables** [comment]
Forgot to release hold in previous comment:
/hold cancel

**Comment by chris-short** [comment]
Added test for @thockin and updated README after re-reading the right one. It's raining readmes this morning.

File: k8s.io/configmap-nginx.yaml (hunk 1 above)
Comment (mrbobbytables) [reply k8s.io/configmap-nginx.yaml]: Thinking about it -- I do kind of like the idea of `yt.k8s.io`/ `youtube.k8s.io` -- the default would go to the landing page for kubernetes, and it opens the door to adding more youtube specific redirects. I'm thinking something like:
`yt.k8s.io/sig-release` -> sig release playlist.

That winds up being a lot friendlier and easier to link to than this:
https://www.youtube.com/playlist?list=PL69nYSiGNLP3QKkOsDsO6A0Y1rhgP84iZ

Comment (chris-short) [reply k8s.io/configmap-nginx.yaml]: I'm down for that! That does make for a nice clean experience between various platforms (GitHub, YouTube, etc.). I have a to do list item for this.

**Comment by munnerz** [comment]
I may be mistaken, but I think this will also require updates to the DNS config for both k8s.io and kubernetes.io to point it at the redirector service: https://github.com/kubernetes/k8s.io/blob/master/dns/zone-configs/k8s.io.yaml

https://github.com/kubernetes/k8s.io/blob/6e4beb92c9798dd85b3457a3f4ff2ee670502395/dns/zone-configs/k8s.io.yaml#L141-L144

**Comment by dims** [comment]
/hold 

looks like there's some more stuff to be done here per @munnerz

**Comment by stp-ip** [comment]
There are no new DNS entries needed currently, but it might be necessary once we move to the redirector. As this is still running in the old cluster it would need migration to either the new cluster or the new redirector anyway.

Will work on pushing the redirector forward so that we might start right with the new redirector depending on how pressing this vanity url is.

**Comment by chris-short** [comment]
So what should my next steps be? I thought I found where DNS entries were
needed Before KubeCon
-- 

Chris Short
He/Him/His
Web <https://chrisshort.net/> | Newsletter <https://devopsish.com/> |
Podcast <http://podcast.podctl.com/> | Twitter
<https://twitter.com/ChrisShort> | LinkedIn
<https://linkedin.com/in/thechrisshort>

**Comment by stp-ip** [comment]
Missed the use of sub domains. This needs a few CNAMEs to point to he current redirector (nginx) you already configured. According to your entries that would be: 

https://github.com/kubernetes/k8s.io/blob/master/dns/zone-configs/k8s.io.yaml
```
youtube:
  type: CNAME
  value: redirect.k8s.io.
yt:
  type: CNAME
  value: redirect.k8s.io.
```

https://github.com/kubernetes/k8s.io/blob/master/dns/zone-configs/kubernetes.io.yaml
```
youtube:
  type: CNAME
  value: redirect.k8s.io.
```

**Pull request closed without merging**